examples/megaclc/megaclc.cpp

```diff
@@ -1606,6 +1606,56 @@ void exec_getchatcallsids(ac::ACState&)
 
 #endif
 
+class OneShotRequestListener : public m::MegaRequestListener
+{
+public:
+    std::function<void(m::MegaApi* api, m::MegaRequest *request)> onRequestStartFunc;
+    std::function<void(m::MegaApi* api, m::MegaRequest *request, m::MegaError* e)> onRequestFinishFunc;
+    std::function<void(m::MegaApi*api, m::MegaRequest *request)> onRequestUpdateFunc;
+    std::function<void(m::MegaApi *api, m::MegaRequest *request, m::MegaError* error)> onRequestTemporaryErrorFunc;
+
+    void onRequestStart(m::MegaApi* api, m::MegaRequest *request) override
+    {
+        if (onRequestStartFunc) onRequestStartFunc(api, request);
+    }
+
+    void onRequestFinish(m::MegaApi* api, m::MegaRequest *request, m::MegaError* e) override
+    {
+        if (onRequestFinishFunc) onRequestFinishFunc(api, request, e);
+        delete this;  // one-shot is done so auto-delete
+    }
+
+    void onRequestUpdate(m::MegaApi*api, m::MegaRequest *request) override
+    {
+        if (onRequestUpdateFunc) onRequestUpdateFunc(api, request);
+    }
+
+    void onRequestTemporaryError(m::MegaApi *api, m::MegaRequest *request, m::MegaError* error) override
+    {
+        if (onRequestTemporaryErrorFunc) onRequestTemporaryErrorFunc(api, request, error);
+    }
+};
+
+
+void exec_apiurl(ac::ACState& s)
+{
+    if (g_megaApi->isLoggedIn())
```

Comment:
The API_URL can be changed even once logged in. All you need to do, in order to keep consistency of data in cache, is a new `fetchnodes()`, so the current cache is discarded and a new fresh one is created from server's data